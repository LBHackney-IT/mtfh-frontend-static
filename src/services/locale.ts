const locale = {
  backButton: "Back",
  processesMenu: {
    title: "Processes",
  },
};

export default locale;
