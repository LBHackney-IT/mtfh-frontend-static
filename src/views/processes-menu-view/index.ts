export * from "./processes-menu-view";
