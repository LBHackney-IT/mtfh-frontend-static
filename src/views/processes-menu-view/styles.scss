@import "~lbh-frontend/lbh/base";

.mtfh-static-processes-menu {
  @include lbh-body-m();
  &__description {
    max-width: 600px;
  }
  &__list {
    list-style: none;
  }
}
