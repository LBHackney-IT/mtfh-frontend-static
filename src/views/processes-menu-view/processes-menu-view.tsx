import { Link as RouterLink, useParams } from "react-router-dom";

import { locale } from "../../services";

import { Heading, Layout, Link } from "@mtfh/common/lib/components";
import "./styles.scss";

const menuItems = [
  {
    label: "Changes to a tenancy",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform",
  },
  {
    label: "Record a tenancy / audit homecheck",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform",
  },
  {
    label: "Find legal referral response",
    link: "https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140",
  },
  {
    label: "Request a legal referral ",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform",
  },
  {
    label: "Record notice served",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform",
  },
  {
    label: "Rehousing",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform",
  },
  {
    label: "Person",
    link: "https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform",
  },
  {
    label: "Report Japanese knotweed",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628",
  },
  {
    label: "Report overgrown tree",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628",
  },
  {
    label: "Report overgrown garden",
    link: "https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628",
  },
];

export const ProcessesMenuView = () => {
  const { id, entityType } = useParams<{ id: string; entityType: string }>();

  return (
    <Layout data-testid="processes-menu" className="mtfh-static-processes-menu">
      <Link as={RouterLink} to={`/${entityType}/${id}`} variant="back-link">
        {locale.backButton}
      </Link>
      <Heading as="h1">{locale.processesMenu.title}</Heading>
      <div className="mtfh-static-processes-menu__description">
        {locale.processesMenu.description()}
      </div>
      <ul className="mtfh-static-processes-menu__list">
        {menuItems.map((item, index) => (
          <li key={index}>
            <Link variant="link" href={item.link} isExternal>
              {item.label}
            </Link>
          </li>
        ))}
      </ul>
    </Layout>
  );
};
