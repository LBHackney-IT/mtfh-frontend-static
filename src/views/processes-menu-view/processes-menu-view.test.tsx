import { render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";

import { locale } from "../../services";
import { ProcessesMenuView } from "./processes-menu-view";

test("it renders processes menu view correctly", () => {
  const { container } = render(<ProcessesMenuView />);
  expect(screen.getByText(locale.backButton));
  expect(screen.getByText(locale.processesMenu.title));
  expect(container).toMatchSnapshot();
});
