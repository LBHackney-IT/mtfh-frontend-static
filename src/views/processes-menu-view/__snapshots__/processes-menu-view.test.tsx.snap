// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders processes menu view correctly 1`] = `
<div>
  <div
    class="mtfh-layout mtfh-layout--narrow mtfh-static-processes-menu"
    data-testid="processes-menu"
  >
    <div
      id="content"
    />
    <div
      class="mtfh-layout__container"
    >
      <div
        class="mtfh-layout__main"
      >
        <a
          class="govuk-back-link lbh-back-link lbh-link--no-visited-state"
          href="/undefined/undefined"
        >
          Back
        </a>
        <h1
          class="lbh-heading-h1"
        >
          Processes
        </h1>
        <p
          class="mtfh-static-processes-menu__description"
        >
          Start a process by selecting a process category below. 
          <br />
          Where a process is not yet supported in 
          <strong>
            Manage My Home
          </strong>
           these links will open the appropriate interim Google form.
        </p>
        <ul
          class="mtfh-static-processes-menu__list"
        >
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSdgJ9DSgGI0Aj7GO1bzLbbrArPabjS8DQwmvwb9ltB-qYYESA/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Changes to a tenancy
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSfyXscVvgCBhbBBe7ADpV6cUqJoKrrTbU0YYRB_uXS2J4AHMQ/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Record a tenancy / audit homecheck
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/spreadsheets/d/1fd6FwBXvBo2DmX8qZaXUr7hX9sE6Zr7HZsPlA_OO8Hs/edit#gid=1251146140"
              rel="noopener noreferrer"
              target="_blank"
            >
              Find legal referral response
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLScpHsoJQgq112HIMbW06YAFpggijwQuODJM483HGbX_PMfCMw/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Request a legal referral 
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSfRSMia4FeT6WPjrmRlsLpOXyJ4tFs3Sv_O0toxUbv_FmIjKw/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Record notice served
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLScPXSs4-xxTH2N5dUwtfmEl5mpu7kxZ6mHe86Qy_rlDNu-GbA/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Rehousing
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLScxrWFLF7M7A3B1yzY-LZFXkV0P0qv6igcQ4wEgWHQk8DaUCw/viewform"
              rel="noopener noreferrer"
              target="_blank"
            >
              Person
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSceU2H-VWi2ewaEiOAkOWULVFU2DoPSen8SRPyJrmQtRpEhfQ/viewform?gxids=7628"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report Japanese knotweed
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSfLmlLraDyOfqYhdjkOfkNWWlHUUep05G5n0GN-ewizN0Fbuw/viewform?gxids=7628"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report overgrown tree
            </a>
          </li>
          <li>
            <a
              class="govuk-link lbh-link"
              href="https://docs.google.com/forms/d/e/1FAIpQLSc3mOUUEQhdRiDsS0k9XDWlfGJG0CEFAC2rnC1sYmM-idt-Nw/viewform?gxids=7628"
              rel="noopener noreferrer"
              target="_blank"
            >
              Report overgrown garden
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
